<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LevelEditor</class>
 <widget class="QMainWindow" name="LevelEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Editor de Niveles de Oakhill</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="toolbox_container" native="true">
       <property name="minimumSize">
        <size>
         <width>400</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>400</width>
         <height>16777215</height>
        </size>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Objetos de la Zona</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Seleccionar Zona:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="combo_zone_selector"/>
           </item>
           <item>
            <widget class="Line" name="line">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QListWidget" name="list_objects"/>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QPushButton" name="btn_add_object">
               <property name="text">
                <string>+ Añadir Objeto</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_delete_object">
               <property name="text">
                <string>- Borrar Objeto</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QScrollArea" name="scrollArea">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>-190</y>
             <width>378</width>
             <height>705</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QGroupBox" name="properties_box">
              <property name="title">
               <string>Propiedades</string>
              </property>
              <layout class="QFormLayout" name="formLayout">
               <item row="0" column="0">
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>ID:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLineEdit" name="prop_id">
                 <property name="readOnly">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_6">
                 <property name="text">
                  <string>Tipo Objeto:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QComboBox" name="prop_type">
                 <item>
                  <property name="text">
                   <string>Obstacle</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Interactable</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string>Posición (X, Y):</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <layout class="QHBoxLayout" name="horizontalLayout_4">
                 <item>
                  <widget class="QSpinBox" name="prop_x">
                   <property name="maximum">
                    <number>9999</number>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="prop_y">
                   <property name="maximum">
                    <number>9999</number>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>Image Path:</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QComboBox" name="prop_image_path_combo">
                 <property name="editable">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="4" column="0">
                <widget class="QLabel" name="label_12">
                 <property name="text">
                  <string>Preview:</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="1">
                <widget class="QLabel" name="prop_image_preview">
                 <property name="minimumSize">
                  <size>
                   <width>0</width>
                   <height>100</height>
                  </size>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">background-color: rgb(22, 22, 22);
border: 1px solid gray;</string>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignCenter</set>
                 </property>
                </widget>
               </item>
               <item row="5" column="0">
                <widget class="QLabel" name="label_8">
                 <property name="text">
                  <string>Resize Factor:</string>
                 </property>
                </widget>
               </item>
               <item row="5" column="1">
                <widget class="QDoubleSpinBox" name="prop_resize_factor">
                 <property name="singleStep">
                  <double>0.100000000000000</double>
                 </property>
                 <property name="value">
                  <double>4.000000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="6" column="0">
                <widget class="QLabel" name="label_9">
                 <property name="text">
                  <string>Opciones:</string>
                 </property>
                </widget>
               </item>
               <item row="6" column="1">
                <layout class="QHBoxLayout" name="horizontalLayout_7">
                 <item>
                  <widget class="QCheckBox" name="prop_is_passable">
                   <property name="text">
                    <string>Es Pasable</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QCheckBox" name="prop_starts_hidden">
                   <property name="text">
                    <string>Empezar Oculto</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item row="7" column="0">
                <widget class="QLabel" name="label_10">
                 <property name="text">
                  <string>Hitbox Offset:</string>
                 </property>
                </widget>
               </item>
               <item row="7" column="1">
                <layout class="QHBoxLayout" name="horizontalLayout_6">
                 <item>
                  <widget class="QSpinBox" name="prop_hitbox_dx">
                   <property name="minimum">
                    <number>-999</number>
                   </property>
                   <property name="maximum">
                    <number>999</number>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="prop_hitbox_dy">
                   <property name="minimum">
                    <number>-999</number>
                   </property>
                   <property name="maximum">
                    <number>999</number>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="prop_hitbox_dw">
                   <property name="minimum">
                    <number>-999</number>
                   </property>
                   <property name="maximum">
                    <number>999</number>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSpinBox" name="prop_hitbox_dh">
                   <property name="minimum">
                    <number>-999</number>
                   </property>
                   <property name="maximum">
                    <number>999</number>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="group_animation">
              <property name="title">
               <string>Animación (Opcional)</string>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_5">
               <item>
                <widget class="QListWidget" name="prop_anim_list"/>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_9">
                 <item>
                  <widget class="QPushButton" name="btn_anim_add">
                   <property name="text">
                    <string>+</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="btn_anim_remove">
                   <property name="text">
                    <string>-</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_10">
                 <item>
                  <widget class="QLabel" name="label_13">
                   <property name="text">
                    <string>Velocidad (Anim Speed):</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QDoubleSpinBox" name="prop_anim_speed">
                   <property name="decimals">
                    <number>3</number>
                   </property>
                   <property name="singleStep">
                    <double>0.010000000000000</double>
                   </property>
                   <property name="value">
                    <double>0.100000000000000</double>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="group_interaction">
              <property name="title">
               <string>Interacción (Solo para Interactables)</string>
              </property>
              <layout class="QFormLayout" name="formLayout_2">
               <item row="0" column="0">
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>Tipo Interacción:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QComboBox" name="prop_interaction_type">
                 <item>
                  <property name="text">
                   <string>None</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Note</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Image</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Door</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_11">
                 <property name="text">
                  <string>Flash Image:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QComboBox" name="prop_flash_image_path_combo">
                 <property name="editable">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>Datos Interacción:</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QStackedWidget" name="prop_interaction_data_stack">
                 <property name="currentIndex">
                  <number>1</number>
                 </property>
                 <widget class="QWidget" name="stack_page_note">
                  <layout class="QVBoxLayout" name="verticalLayout_3">
                   <item>
                    <widget class="QTextEdit" name="data_note_text"/>
                   </item>
                  </layout>
                 </widget>
                 <widget class="QWidget" name="stack_page_image">
                  <layout class="QHBoxLayout" name="horizontalLayout_2">
                   <item>
                    <widget class="QComboBox" name="data_image_path_combo">
                     <property name="editable">
                      <bool>true</bool>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                 <widget class="QWidget" name="stack_page_door"/>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QGraphicsView" name="canvas_view"/>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuArchivo">
    <property name="title">
     <string>Archivo</string>
    </property>
    <addaction name="action_load_json"/>
    <addaction name="action_save_json"/>
   </widget>
   <addaction name="menuArchivo"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_load_json">
   <property name="text">
    <string>Cargar JSON</string>
   </property>
  </action>
  <action name="action_save_json">
   <property name="text">
    <string>Guardar JSON</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>